extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/KinesisRecordSupplier.java

```diff
@@ -294,22 +296,18 @@ private Runnable fetchRecords()
 
             // If the buffer was full and we weren't able to add the message, grab a new stream iterator starting
             // from this message and back off for a bit to let the buffer drain before retrying.
-            if (!records.offer(currRecord, recordBufferOfferTimeout, TimeUnit.MILLISECONDS)) {
```

Comment:
The comment above is no longer accurate -- we aren't grabbing new stream iterators anymore when the buffer is full.

Reply:
removed.